kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/model/networking/LoadBalancerClusterIngressNetworkingModel.java

```diff
@@ -65,6 +74,11 @@ public boolean requiresSharedSniContainerPort() {
 
     @Override
     public Stream<Integer> requiredSniLoadBalancerServicePorts() {
-        return Stream.of(DEFAULT_LOADBALANCER_PORT);
+        return Stream.of(DEFAULT_CLIENT_FACING_LOADBALANCER_PORT);
+    }
+
+    @Override
+    public String bootstrapServers() {
+        return new HostPort(loadBalancer.getBootstrapAddress(), DEFAULT_CLIENT_FACING_LOADBALANCER_PORT).toString().replace("$(virtualClusterName)", name(cluster));
```

Comment:
This uses `HostPort` just to format the string. So would 
```suggestion
        return String.format("%s:%s", loadBalancer.getBootstrapAddress(), DEFAULT_CLIENT_FACING_LOADBALANCER_PORT).replace("$(virtualClusterName)", name(cluster));
```

Be easier and more obvious?

Reply:
Maybe the signature should be HostPort, I think we should favour it as much as we can and convert to String later.

Reply:
Sounds like a good start, what about the token replacement should we have an extension of or wrapper for `HostPort` that understands the replacements ?

Reply:
Yep it would be nice if we didn't have to duplicate the logic done in the Proxy.

Some deep bit of the runtime is currently in control of the replacements, ` io.kroxylicious.proxy.internal.clusternetworkaddressconfigprovider.BrokerAddressPatternUtils`. I think the literal values at least should be put into `kroxylicious-api` somewhere, maybe the code to do the substitutions into a String too.

Will leave that for another PR I think